tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py

```diff
@@ -0,0 +1,338 @@
+    create_ceph_block_pool.__name__
+)
+@pytest.mark.polarion_id("OCS-682")
+class TestPVSTOcsCreateDeletePVCsWithoutIO(BasePvcPodCreateDelete):
+    """
+    Test OCS-682 Create & Delete Cluster with 1000 PVC without any IO,
+    then Increase the PVC count to 1500. Check for Memory leak
+    """
+
+    @pytest.fixture()
+    def ocs682_fixture(self, request):
+        def finalizer():
+            self.cleanup()
+
+        request.addfinalizer(finalizer)
+
+    def test_pvst_create_delete_pvcs_without_io(self, memory_leak_function, ocs682_fixture):
+        self.pvc_count_1st_itr = 5
+        self.pvc_count_next_itr = 10
+        self.scale_pod_count = 120
+        self.test_case = 'ocs682'
+        self.size = '10Gi'
+
+        log.info(f"Test execution polation ID OCS-682")
+
+        self.pre_req()
+        # Creating sc objs in need to use both RBD & CephFS together in TC
+        self.rbd_sc_obj = helpers.create_storage_class(
+            interface_type=constants.CEPHBLOCKPOOL,
+            interface_name=self.cbp_obj.name,
+            secret_name=self.rbd_secret_obj.name)
+        self.cephfs_sc_obj = helpers.create_storage_class(
+            interface_type=constants.CEPHFILESYSTEM,
+            interface_name=helpers.get_cephfs_data_pool_name(),
+            secret_name=self.cephfs_secret_obj.name)
+        log.info(f"Created sc obj for cephfs {self.cephfs_sc_obj} "
+                 f"and rbd {self.rbd_sc_obj}")
+        # First Iteration call to create PVC and POD
+        self.create_pvc_pod(self.rbd_sc_obj, self.cephfs_sc_obj, self.pvc_count_1st_itr,
+                            self.size)
+        # Continue to iterate till the scale pvc limit is reached
+        # Also continue to perform create and delete of pod, pvc in parallel
+        while True:
+            if self.scale_pod_count <= len(self.all_pod_obj):
+                log.info(f"Created {self.scale_pod_count} pvc and pods")
+                break
+            else:
+                with ThreadPoolExecutor() as executor:
+                    log.info(f"Create {self.pvc_count_next_itr} and "
+                             f"in parallel delete {self.delete_pod_count} "
+                             f"pods & pvc")
+                    thread_list = [self.delete_pvc_pod(), self.create_pvc_pod(
+                        self.rbd_sc_obj, self.cephfs_sc_obj, self.pvc_count_next_itr,
+                        self.size)]
+                    for thread in thread_list:
+                        executor.submit(thread)
+        # Added sleep for the test case to wait for capturing memory leak after scale
+        time.sleep(self.timeout)
+        helpers.memory_leak_analysis(self.median_dict)
+
+
+@scale
+@pytest.mark.usefixtures(
+    create_project.__name__,
+    create_rbd_secret.__name__,
+    create_cephfs_secret.__name__,
+    create_ceph_block_pool.__name__
+)
+@pytest.mark.polarion_id("OCS-679")
+class TestPVSTOcsCreateDeletePVCsWithIO(BasePvcPodCreateDelete):
```

Comment:
the implementation in this test class and the one above looks pretty much similar. Please try to use pytest parametrize and combine both under the same class

Reply:
Tried to use parametrize but observing below problem 
memory_leak_function is not stopped in the teardown part of first param execution, is this expected ?. At end of test observing ResourceLeftoversException

Reply:
Implemented parametrize resolving this comment.